core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1108,6 +1137,18 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def sandbox: ZIO[R, Cause[E], A] = foldCauseM(ZIO.fail, ZIO.succeed)
 
+  /**
+   * Extracts the optional value, or fails with the given error 'f'.
+   */
+  def someOrFail[B, E1 >: E](f: E1)(implicit ev: A <:< Option[B]): ZIO[R, E1, B] =
+    self.flatMap(ev(_).fold[ZIO[R, E1, B]](ZIO.fail(f))(ZIO.succeed))
+
+  /**
+   * Extracts the optional value, or fails with a [[NoSuchElementException]]
+   */
+  def someOrFailException[B, E1 >: NoSuchElementException](implicit ev: A <:< Option[B], ev2: E <:< E1): ZIO[R, E1, B] =
```

Comment:
Same comments as with `leftOrFailException`

Reply:
This method does almost the same as `ZIO#get`, do we really need both?